src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/TransactionServiceTest.java

```diff
@@ -0,0 +1,86 @@
+package uk.gov.companieshouse.overseasentitiesapi.service;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.api.ApiClient;
+import uk.gov.companieshouse.api.error.ApiErrorResponseException;
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
+import uk.gov.companieshouse.api.handler.transaction.TransactionsResourceHandler;
+import uk.gov.companieshouse.api.handler.transaction.request.TransactionsGet;
+import uk.gov.companieshouse.api.model.ApiResponse;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+import uk.gov.companieshouse.overseasentitiesapi.client.ApiClientService;
+import uk.gov.companieshouse.overseasentitiesapi.exception.ServiceException;
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class TransactionServiceTest {
+
+    private static final String TRANSACTION_ID = "12345678";
+    private static final String PASSTHROUGH_HEADER = "passthrough";
+
+    @Mock
+    private ApiClientService apiClientService;
+
+    @Mock
+    private ApiClient apiClient;
+
+    @Mock
+    private TransactionsResourceHandler transactionsResourceHandler;
+
+    @Mock
+    private TransactionsGet transactionsGet;
+
+    @Mock
+    private ApiResponse<Transaction> apiResponse;
+
+    @InjectMocks
+    private TransactionService transactionService;
+
+    @Test
+    void getTransaction() throws ServiceException, IOException, URIValidationException {
```

Comment:
In recent years I've been naming these methods so as to describe what is being tested... e.g. this one could be 'testGettingATransactionIsSuccessful()' and, for example, method below 'testServiceExceptionThrownWhenTransactionRetrievalUnsuccessful' (if that's what it's testing).

Up to you. No reason not to approve...

Reply:
i can update them